frontend/src/components/P4Editor.tsx

```diff
@@ -3,64 +3,200 @@ import Button from "@material-ui/core/Button";
+
+type Severity = "error" | "success" | "info" | "warning" | undefined;
+
 interface State {
   code: string
+  editorResult: string
+  editorSeverity: Severity
+  editorNotificationOpen: boolean
+  fileChanged: boolean
+  editorConfirmationDialogOpen: boolean
 }
 
 interface P4EditorProps {
   endpoint: string;
 }
 
+function Alert(props: JSX.IntrinsicAttributes & AlertProps) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
 export default class P4Editor extends React.Component<P4EditorProps> {
   public state: State;
   private editor!: monaco.editor.IStandaloneCodeEditor;
-  
+
   constructor(props: P4EditorProps) {
     super(props);
     this.state = {
       code: "",
+      editorResult: "",
+      editorSeverity: "error",
+      editorNotificationOpen: false,
+      fileChanged: false,
+      editorConfirmationDialogOpen: false
     };
     this.save = this.save.bind(this);
+    this.load = this.load.bind(this);
   }
 
-  editorDidMount(editor: monaco.editor.IStandaloneCodeEditor){
+  editorDidMount(editor: monaco.editor.IStandaloneCodeEditor) {
     editor.focus();
     this.editor = editor;
-    fetch(APIRequest(`${this.props.endpoint}`, {headers: {'Content-Type': 'application/json', authorization: localStorage.getItem("token") || ""} }))
+    fetch(APIRequest(`${this.props.endpoint}`, { headers: { 'Content-Type': 'application/json', authorization: localStorage.getItem("token") || "" } }))
       .then((response) => response.text())
       .then((data) => {
         this.setState({ code: data });
       })
       .catch((err) => console.error(err));
   };
 
-  onChange(newValue: string){
-    this.setState({code: newValue})
+  onChange(newValue: string) {
+    this.setState({ code: newValue })
+    this.setState({ fileChanged: true })
   };
 
-  save(): void {
-    fetch(APIRequest(this.props.endpoint, {
-      method: "post", 
-    body: this.editor.getModel()?.getValue(),headers: {'Content-Type': 'text/plain', 
-    authorization: localStorage.getItem("token") || ""}  }))
-      .then((response) => response.text())
-      .catch((error) => {
-        console.error(error);
-      });
+  async save(): Promise<void> {
+    this.setState({ fileChanged: false })
+    this.setState({ editorResult: "Saving file..." })
+    this.setState({ editorSeverity: "info" })
+    this.setState({ editorNotificationOpen: true })
+    try {
+      const result = await fetch(APIRequest(this.props.endpoint, {
+        method: "post",
+        body: this.editor.getModel()?.getValue(), headers: {
+          'Content-Type': 'text/plain',
+          authorization: localStorage.getItem("token") || ""
+        }
+      }))
+      if (result.status === 200) {
+        this.setState({ editorResult: "Deploy successful!" })
+        this.setState({ editorSeverity: "success" })
+        this.setState({ editorNotificationOpen: true })
+        this.setState({ fileChanged: false })
+      }
+      else {
+        const a = await result.json()
```

Comment:
"a" is a very bad variable name

Reply:
absolutely ;) I found another var name "a" in LoginForm.tsx and also refactored that one